src/basic/time-util.c

```diff
@@ -1605,19 +1605,29 @@ bool clock_supported(clockid_t clock) {
 }
 
 int get_timezone(char **ret) {
-        _cleanup_free_ char *t = NULL;
+        _cleanup_free_ char *t = NULL, *link_dir = NULL, *abspath = NULL;
         int r;
 
         assert(ret);
 
-        r = readlink_malloc("/etc/localtime", &t);
+        const char *link_path = etc_localtime();
+
+        r = readlink_malloc(link_path, &t);
         if (r == -ENOENT)
                 /* If the symlink does not exist, assume "UTC", like glibc does */
                 return strdup_to(ret, "UTC");
         if (r < 0)
                 return r; /* Return EINVAL if not a symlink */
 
-        const char *e = PATH_STARTSWITH_SET(t, "/usr/share/zoneinfo/", "../usr/share/zoneinfo/");
+        r = path_extract_directory(link_path, &link_dir);
+        if (r < 0)
+                return r;
+
+        abspath = path_make_absolute(t, link_dir);
+        if (!abspath)
+                return -ENOMEM;
+
+        const char *e = path_startswith(abspath, "/usr/share/zoneinfo/");
```

Comment:
Hm? I cannot follow the logic.
At least, previously we support symlink with relative path. But this drops the support...